File: healthchecks/order-service/src/main/java/com/varunu28/orderservice/repository/OrderRepository.java
package com.varunu28.orderservice.repository;

import com.varunu28.orderservice.model.Order;
import java.util.UUID;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface OrderRepository extends CrudRepository<Order, UUID> {
}

File: saga/order-service/src/main/kotlin/com/saga/orderservice/OrderRepository.kt
package com.saga.orderservice

import jakarta.persistence.LockModeType
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Lock
import org.springframework.stereotype.Repository
import java.math.BigInteger

@Repository
interface OrderRepository : JpaRepository<Order, BigInteger> {

    @Lock(LockModeType.PESSIMISTIC_WRITE)
    fun findOrderById(id: BigInteger): Order?
}